engine/src/history.rs

```diff
@@ -37,6 +37,23 @@ impl History {
         }
     }
 
+    pub fn new_from_str(moves: String) -> Result<Self, GameError> {
+        let mut history = History::new();
+        for mov in moves.split(";") {
+            let split = mov.split_whitespace().collect::<Vec<&str>>();
+            let piece = split.get(0).ok_or(GameError::ParsingError {
+                found: "NA".to_string(),
+                typ: "Piece".to_string(),
+            })?;
+            let pos = split.get(1).ok_or(GameError::ParsingError {
+                found: "NA".to_string(),
+                typ: "Position".to_string(),
+            })?;
+            history.moves.push((piece.to_string(), pos.to_string()));
```

Comment:
imo `History` should really store validated Piece and Position types, not strings